linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -1830,10 +1774,39 @@ private[emitter] object CoreJSLib {
                   typedArrayClass))
         }))
       }
+
+      Block(obj :: prims)
     }
 
-    private def defineFunction(name: String, args: List[ParamDef], body: Tree): Unit =
-      buf += extractWithGlobals(globalFunctionDef(name, CoreVar, args, body))
+    private def defineFunction(name: String, args: List[ParamDef], body: Tree): Tree =
+      extractWithGlobals(globalFunctionDef(name, CoreVar, args, body))
+
+    private val argRefs = Stream.from(1).map(i => varRef("arg" + i))
```

Comment:
`Stream` is deprecated in 2.13, so we can't use it. Anyway, it seems to me that the laziness is not really necessary, and that using a List with a fixed size would be more straightforward and efficient:
```suggestion
    private val argRefs = List.tabulate(5)(i => varRef("arg" + i))
```